### DISCUSSION TITLE
Please help! Code for USB power up and hindsight with motion dedection wanted.

### DISCUSSION THREAD

**Discussion by Hubsi4** [gopro/labs#540] [Q&A] [answered]
I would like to use my Hero11black with Labs 02.10.70 for hours of automatic recording without an internal battery, only with USB power from a large power bank. Since I'm a newbie and have no experience with programming and am a bit overwhelmed, I'm asking you for help. Maybe someone would be so kind and can generate the following code for me in one go: 
power up/boot on USB Power (without battery!), set date and time, set mode to video, start hindsight (15 sec) and motion dedection (scene, delay start 300 sec, start sensitivity 4, mask motion 1sec, hold time 60 sec, stop sensitivity 5, repeat)
Many Thanks!

**Comment by dnewman-gpsw** [comment]
Not fully possible, where would the time and date come from?  Battery removed you will lose data/time, and the QR Code can only a static data/time, so you can't have a repeating command that sets correct time.

Otherwise use:
    !MWAKE=2  <- wake on power.
    hS1!SM4-5D300M1H60!R  <- your weird motion settings.  Why delay 300 seconds?

**Reply by Hubsi4** [reply]
Thanks! Then without set time and date, I can do it with my smartphone. Do you know, how long date and time is in memory when battery is removed. Is it immediately gone?
I need 5 min to leave the scene, the cam is lifted on a pole and many other things have to be done.

**Reply by dnewman-gpsw** [reply]
Clock without battery is good for 12-24 hours.

**Reply by Hubsi4** [reply]
I tried now, but unfortunately the wake on power code does not work. Also motion dedection does not work, only hindsight.
I would like a code that is permanently stored, so on every USB Power on, the camera boots and starts hindsight and motion dedection together and automatically, without showing QR code.

**Reply by Hubsi4** [reply]
With battery installed, the wake on USB Power works, but without battery it does not work. Also hindsight and motion dedection does not start, with or without battery. I have heat problems, so it is mandatory, to run the cam without battery.

**Reply by dnewman-gpsw** [reply]
Look up using boot commands. You might need TUSB to fix your power issues.

**Reply by Hubsi4** [reply]
TUSB I already use. Power is not an issue. How is the code for this boot commands to start hindsight and motion dedection at the same time? I am not familiar with programming. Please help.

**Reply by dnewman-gpsw** [reply]
Your camera will boot without a battery with a good external USB power, if !MWAKE=2 is set. I have confirmed this here. 

Your motion detect as a boot command:    !MBOOT="!Lbt"!SAVEbt=hS1!SM4-5D300M1H60!R

**Reply by Hubsi4** [reply]
No, my camera does not boot on USB power with !MWAKE=2 without battery. But i can boot by manualy pressing mode button, so USB power is enough. My Powerbank delivers 5V 3.0A 15Watts. I have also activated !MTUSB=1
When manually boot, motion detect works, with  !MBOOT="!Lbt"!SAVEbt=hS1!SM4-5D300M1H60!R, but obviously it kills hindsight, so only motion detect works.

**Reply by Hubsi4** [reply]
How can I achieve motion detection together with hindsight?

**Reply by dnewman-gpsw** [reply]
Are you running current firmware?   As all this is working today here.  It seems you are on 2.10.70, not 2.12.70.

**Reply by Hubsi4** [reply]
Oh yes, I am still on 2.10.70, sorry. I will try immediately!

**Reply by Hubsi4** [reply]
Now I am running 2.12.70, all labs reset, only !MTUSB=1 and !MWAKE=2 But still the same. No boot on USB Power without battery. I'm desperate.

**Reply by Hubsi4** [reply]
I have two Hero11black, both run 2.12.70 now and show same behavior. No automatic boot on USB power without battery. With battery inside, boot works on plug in USB power, but that is not what I need.

**Reply by dnewman-gpsw** [reply, marked as answer]
I discovered there was a minor patch for battery-less WAKEs a few weeks ago. Sorry I has forgotted and I was running this patch. There was a new build with that patch https://www.dropbox.com/t/UfbTGbMf8W6SsTLw

**Reply by Hubsi4** [reply]
Thank you very much, David! Everything is working now as expected!
You safed my life! I am working on a tv wildlife project and had no idea to do it without this features.
Gopro Labs rocks!!
